src/ride_gui.cpp

```diff
@@ -129,10 +129,7 @@ RideSelectGui::RideSelectGui() : GuiWindow(WC_RIDE_SELECT, ALL_WINDOWS_OF_TYPE)
 	this->SetNewRideKind(this->current_kind, true);
 }
 
-RideSelectGui::~RideSelectGui()
-{
-	_rides_manager.CheckNoAllocatedRides(); // Check that no rides are being constructed.
-}
+RideSelectGui::~RideSelectGui() = default;
```

Comment:
Just remove this line and the `~RideSelectGui()` in the class declaration? (Or set it to `default` there? not sure if there is any difference, I'd expect not.)

Reply:
Moved it to the class declaration. I've heard some compilers can optimize better if ctors and dtors are explicitly defaulted than if they're missing or have an empty body.